components/kyma-environment-broker/internal/lms/client.go

```diff
@@ -0,0 +1,398 @@
+package lms
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	random "math/rand"
+	"net/http"
+
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/iosafety"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Client interface {
+	CreateTenant(input CreateTenantInput) (o CreateTenantOutput, err error)
+	GetTenantStatus(tenantID string) (status TenantStatus, err error)
+	GetTenantInfo(tenantID string) (status TenantInfo, err error)
+
+	GetCACertificate(tenantID string) (cert string, found bool, err error)
+	GetSignedCertificate(tenantID string, certID string) (cert string, found bool, err error)
+	RequestCertificate(tenantID string, subject pkix.Name) (string, []byte, error)
+}
+
+type Environment string
+
+type ClusterType string
+
+type client struct {
+	url string
+
+	// tenant predefined values
+	clusterType ClusterType
+	environment Environment
+
+	token      string
+	samlTenant string
+
+	log logrus.FieldLogger
+}
+
+const (
+	EnvironmentDev  Environment = "dev"
+	EnvironmentProd Environment = "prod"
+
+	ClusterTypeHA         ClusterType = "ha"
+	ClusterTypeSingleNode ClusterType = "single-node"
+)
+
+type Config struct {
+	URL string
+
+	// tenant predefined values
+	ClusterType ClusterType // ha or single-node
+	Environment Environment
+
+	Token      string
+	SamlTenant string
+
+	Disabled   bool
+	FakeClient bool
+}
+
+func NewClient(cfg Config, log logrus.FieldLogger) Client {
+	if cfg.FakeClient {
+		return NewFakeClient(15 * time.Second)
```

Comment:
Why is this condition here? fake or not client is controlled by ENV?

Reply:
we can start KEB with fake lms client, but if you think it is not a good idea (makes the code smell) - I cen remove it. It was created for testing purpose

Reply:
I'll remove it